core/src/main/java/ai/timefold/solver/core/api/score/stream/common/ConcurrentUsageInfo.java

```diff
@@ -0,0 +1,16 @@
+package ai.timefold.solver.core.api.score.stream.common;
+
+public interface ConcurrentUsageInfo<Interval_, Point_ extends Comparable<Point_>, Difference_ extends Comparable<Difference_>> {
+
+    /**
+     * @return never null, an iterable that iterates through the interval clusters
+     *         contained in the collection in ascending order
+     */
+    Iterable<ConcurrentUsage<Interval_, Point_, Difference_>> getConcurrentUsages();
```

Comment:
This, in my opinion, is another place where the new name "concurrent usage" doesn't really work. "get concurrent usages" sounds to me a bit like something a non-native speaker would produce.

Overlapping intervals probably wasn't ideal either. How about overlapping ranges?

Reply:
(`connected` is also what the term is called in math, see https://www.math.toronto.edu/ivan/mat327/docs/notes/18-connected.pdf ) (in particular, it tracks the connected components of the space defined by the union of all intervals: https://en.wikipedia.org/wiki/Connected_space#Connected_components )

Reply:
Connected ranges, connected intervals, I like either of them.